app/models/user.rb

```diff
@@ -689,6 +690,10 @@ def ensure_default_role
     self.roles << default_role unless self.roles.include?(default_role)
   end
 
+  def set_default_language_for_new_user
+    self.locale = Setting[:default_language] if self.respond_to?(:locale) && Setting[:default_language].present?
```

Comment:
Is it necessary for some reason to check if we `respond_to?(:locale)`?
```suggestion
    self.locale ||= Setting[:default_language].presence
```
The only important part here for me is not to overwrite the value if there is one.